pkg/terraformer/terraformer.go

```diff
@@ -205,3 +224,63 @@ func (t *Terraformer) executeTerraform(ctx context.Context, command Command) err
 	log.Info("terraform process finished successfully", "command", command)
 	return nil
 }
+
+func (t *Terraformer) addFinalizer(ctx context.Context) error {
+	logger := t.log.WithName("add-finalizer")
+	return t.updateObjectFinalizers(ctx, logger, controllerutils.EnsureFinalizer)
+
+}
+
+func (t *Terraformer) removeFinalizer(ctx context.Context) error {
+	logger := t.log.WithName("remove-finalizer")
+	return t.updateObjectFinalizers(ctx, logger, controllerutils.DeleteFinalizer)
```

Comment:
Because of the usage of `controllerutils.EnsureFinalizes` and `controllerutils.DeleteFinalizer` a lot of dependencies are introduced in the vendor, I am not sure whether have to keep it like it is or to duplicate some code here. 
Any opinion?

Reply:
Just checked and the binary size is increased from 29.1MiB to 39.7MiB with this PR.

Reply:
I managed to switch to patch only from the controller utils module.